src/pyubx2/ubxreader.py

```diff
@@ -109,23 +109,22 @@ def __next__(self) -> tuple:
             raise StopIteration
         return (raw_data, parsed_data)
 
-    def read(self) -> tuple:
+    def read(self, parsing=True) -> tuple:

```

Comment:
This causes the program to hang if you're reading from a dummy socket input (via pyubx2's interal SocketStream wrapper) - try running /tests/test_socket.py.